### PR TITLE
can setting tabnine's min and max value

### PR DESCRIPTION
**Pull request by get-me-power**
Hi, @kitagry !
Thanks for creating nice vim-plugin!

I used global variables to allow some values to be set on a per-user basis.
Could you check this PR?

### CODE REVIEW

**Comment by get-me-power** [comment]
I fixed conflict.

**Comment by kitagry** [comment]
Thanks @kazukazuinaina 

But, I would like to set these variables like followings.This is same as [asyncomplete-buffer.vim](https://github.com/prabirshrestha/asyncomplete-buffer.vim#registration).
So, could you fix it like this?

```vim
call asyncomplete#register_source(asyncomplete#sources#tabnine#get_source_options({
  \ 'name': 'tabnine',
  \ 'allowlist': ['*'],
  \ 'completor': function('asyncomplete#sources#tabnine#completor'),
  \ 'config': {
  \   'line_limit': 1000,
  \   'max_num_result': 20,
  \ }))
```

**Comment by get-me-power** [comment]
@kitagry 
Thanks for review.
How about this changing?

File: autoload/asyncomplete/sources/tabnine.vim [hunk 1]
```diff
@@ -34,14 +32,16 @@ function! s:start_tabnine() abort
 endfunction
 
 function! s:get_response(opt, ctx) abort
+    let l:line_limit = get(a:opt['config'], 'line_limit', 1000)
+    let l:max_num_result = get(a:opt['config'], 'max_num_result', 20)
```
Comment (kitagry) [inline autoload/asyncomplete/sources/tabnine.vim]: Could you change 20 to 10?

Comment (get-me-power) [reply autoload/asyncomplete/sources/tabnine.vim]: Thanks, I fixed.

**Comment by kitagry** [comment]
Thanks @kazukazuinaina

**Comment by get-me-power** [comment]
@kitagry
Thanks for your merging!

**Pull request merged**